frontend/src/metabase/dashboard/components/Dashboard/Dashboard.tsx

```diff
@@ -173,167 +174,230 @@ interface DashboardProps {
-          tabId: this.props.dashboard.tabs?.[0]?.id ?? null,
-        });
-      }
-    } catch (err) {
-      const error = err as any;
-      if (error.status === 404) {
-        setErrorPage({ ...error, context: "dashboard" });
-      } else {
-        console.error(error);
-        this.setState({ error });
-      }
-    }
-  }
+    const node = getMainElement();
 
-  setEditing = (dashboard: IDashboard) => {
-    this.props.onRefreshPeriodChange(null);
-    this.props.setEditingDashboard(dashboard);
-  };
+    const handleScroll = (event: any) => {
+      setHasScroll(event.target.scrollTop > 0);
+    };
 
-  setDashboardAttribute = <Key extends keyof IDashboard>(
-    attribute: Key,
-    value: IDashboard[Key],
-  ) => {
-    this.props.setDashboardAttributes({
-      id: this.props.dashboard.id,
-      attributes: { [attribute]: value },
+    node.addEventListener("scroll", handleScroll, {
+      capture: false,
+      passive: true,
     });
-  };
-
-  onCancel = () => {
-    this.props.setSharing(false);
-  };
-
-  onSharingClick = () => {
-    this.props.setSharing(!this.props.isSharing);
-  };
-
-  onAddQuestion = () => {
-    const { dashboard } = this.props;
-    this.setEditing(dashboard);
-    this.props.toggleSidebar(SIDEBAR_NAME.addQuestion);
-  };
-
-  onMainScroll = (event: any) => {
-    this.setState({ hasScroll: event.target.scrollTop > 0 });
-  };
-
-  renderContent = () => {
-    const { dashboard, selectedTabId, isNightMode, isFullscreen } = this.props;
 
-    const canWrite = dashboard?.can_write ?? false;
+    return () => node.removeEventListener("scroll", handleScroll);
+  }, [isInitialized]);
 
-    const dashboardHasCards = dashboard?.dashcards.length > 0 ?? false;
-
-    const tabHasCards =
-      dashboard?.dashcards.filter(
-        c =>
-          selectedTabId !== undefined && c.dashboard_tab_id === selectedTabId,
-      ).length > 0 ?? false;
+  useEffect(() => {
+    if (!previousDashboard && dashboard) {
+      fetchDashboardCardData({ reload: false, clearCache: true });
+    }
+  }, [dashboard, fetchDashboardCardData, previousDashboard]);
 
-    const shouldRenderAsNightMode = isNightMode && isFullscreen;
+  useEffect(() => {
+    if (isInitialized && previousDashboardId !== dashboardId) {
+      handleLoadDashboard(dashboardId);
+    }
+  }, [dashboardId, handleLoadDashboard, isInitialized, previousDashboardId]);
 
+  useEffect(() => {
+    if (isInitialized && previousTabId !== selectedTabId) {
+      fetchDashboardCardData();
+      fetchDashboardCardMetadata();
+    }
+  }, [
+    fetchDashboardCardData,
+    fetchDashboardCardMetadata,
+    isInitialized,
+    previousTabId,
+    selectedTabId,
+  ]);
+
+  useEffect(() => {
+    if (isInitialized && !_.isEqual(previousParameterValues, parameterValues)) {
+      fetchDashboardCardData({ reload: false, clearCache: true });
+    }
+  }, [
+    fetchDashboardCardData,
+    isInitialized,
+    parameterValues,
+    previousParameterValues,
+  ]);
```

Comment:
:x: 

`componentDidUpdate` got split into 3 `useEffect` hooks.
Notice that previously there were `return` statements in `componentDidUpdate` - we need to migrate this logic as well. Otherwise we will get unnecessary network requests.

So: 2nd hook should not run if the 1st runs, and 3rd one should not run if 1st or 2nd runs.

I suggest to merge these 3 hooks into 1, which would look identical to the old `componentDidUpdate`.

We might be able to get rid of `useMount` hook as well by incorporating it here. This should help avoid unnecessary network requests.